jax/scipy/spatial/distance.py

```diff
@@ -0,0 +1,150 @@
+import jax.numpy as jnp
+from jax import jit, vmap
+
+# define primitives for distance functions according to scipy.spatial.distance
+# which can be batched over the first dimension and then used inside jit and vmap
+# as well as used for implementing cdist and pdist versions
+
+primitives = {
```

Comment:
Rather than a dict of anonymous functions, I'd probably define these as named functions in the `distance` namespace, like `scipy.spatial.distance.braycurtis`, etc. It will make the implementations easier to read.

If it's useful for `pdist` you could then refer to these implementations in a dict.

Also, I'd validate the input shapes as well – the implementations here will return strange results if you pass in broadcast-compatible multi-dimensional arrays